src/instruction.rs

```diff
@@ -0,0 +1,268 @@
+//! Program instruction types.
+
+use solana_program::{
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+
+/// Instructions supported by the Paladin Rewards program.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PaladinRewardsInstruction {
+    /// Configures rewards for a mint that has been configured with the rewards
+    /// program as a transfer hook program.
+    ///
+    /// This instruction will:
+    ///
+    /// - Initialize a rewards account for the mint (distribution account),
+    ///   configured with the distribution addresses:
+    ///   - Active rewards
+    ///   - Piggy bank
+    ///   - Staked PAL rewards
+    /// - Initialize an active rewards account for the mint.
+    /// - Initialize the required accounts for the transfer hook.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    /// 0. `[w]` Distribution account.
+    /// 1. `[w]` Active rewards account.
+    /// 2. `[w]` Transfer hook extra account metas account.
+    /// 3. `[ ]` Token mint.
+    /// 4. `[s]` Mint authority.
+    InitializeMintRewardInfo {
+        piggy_bank_address: Pubkey,
+        staked_rewards_address: Pubkey,
+    },
+    /// Moves the active rewards to the distribution account and updates total
+    /// rewards.
+    ///
+    /// Accounts expected by this instruction:
+    ///
+    /// 0. `[w]` Active rewards account.
+    /// 1. `[w]` Distribution account.
+    /// 2. `[ ]` Token mint.
```

Comment:
I'd lean toward storing it if there's really no reason we need it here. Since it doesn't look like we're going to read it, I'd say it's fine to store.

Reply:
Turns out we're cutting this instruction, so this problem goes away!